eng-wasm/src/crypto_wasm.rs

```diff
@@ -4,54 +4,30 @@ use eng_pwasm_abi::types::U256;
 use rand_wasm::Rand;
 
 const SYMMETRIC_KEY_SIZE: usize = 32;
-const IV_SIZE: usize = 96 / 8;
-const PAYLOAD_SIZE: usize = 1024;
 
-pub type IV = [u8; IV_SIZE];
 pub type SymmetricKey = [u8; SYMMETRIC_KEY_SIZE];
 
+fn extra_size_for_cypher() -> usize {
+    unsafe {(external::get_iv_size() + external:: get_tag_size()) as usize }
```

Comment:
Should comment what this tag_size is and why you need it

Reply:
Done